src/craftdef.h

```diff
@@ -44,6 +44,18 @@ enum CraftMethod
 };
 
 /*
+	The type a hash can be. The earlier a type is mentioned in this enum,
+	the earlier it is tried, and the less likely is a collision
+ */
+enum CraftHashType
+{
+	CRAFT_HASH_TYPE_ITEM_NAMES = 0,
+	CRAFT_HASH_TYPE_COUNT,
+	CRAFT_HASH_TYPE_MAX
+};
+const int craft_hash_type_unhashed = (int) CRAFT_HASH_TYPE_MAX;
```

Comment:
My goal was having two separate variables, one that can be used when looping over all hash types, and one the unhashed craft definitions are stored in. Perhaps I should flip the roles, and make max a const. Do you agree then?